Update SCTP KEP with Graduation Criteria

danwinship opened the pull request:
There don't seem to be any docs on what the KEP `status` values mean, but I assume it shouldn't be `implemented` until it's GA? So I bumped it back to `implementable`.

There are two "FIXME"s to figure out, plus I'm pretty unsure about the beta->GA criteria. (Especially the "2 implementations" part.)

/assign @thockin 
cc @kubernetes/sig-network-misc

thockin commented on keps/sig-network/0015-20180614-SCTP-support.md:
```diff
@@ -297,11 +298,71 @@ We describe in the Kubernetes documentation the mutually exclusive nature of use
+
+  - The test also ensures that if `sctp.ko` was not loaded before the
+    test (on the node with the created test pod), then it is still not
+    loaded after the test. (FIXME: that could fail if it ran in
```
I'm not up on current test capabilities, but can't we mark it as disruptive so it doesn't run in parallel?

thockin commented on keps/sig-network/0015-20180614-SCTP-support.md:
```diff
@@ -297,11 +298,71 @@ We describe in the Kubernetes documentation the mutually exclusive nature of use
+    loaded after the test. (FIXME: that could fail if it ran in
+    parallel with some of the other tests below.)
+
+  - (FIXME: how does e2e work with feature gates? This test and the
```
I actually don't know the answer to this any more.  @BenTheElder can you comment on testing patterns for stuff like this, or refer to someone who knows best?

BenTheElder replied:
it's not super great, we often have a seperate copy of some standard job with the feature gate togggled in the deployer and then including the `[Feature:Foo]` ginkgo test tags.

toggling it will be dependent on the test setup.

imo we could do with a more robust strategy for this, i'm not really sure we have one.

thockin commented on keps/sig-network/0015-20180614-SCTP-support.md:
```diff
@@ -297,11 +298,71 @@ We describe in the Kubernetes documentation the mutually exclusive nature of use
-## Alternatives [optional]
+Graduation criteria:
+
+- At least 2 external network plugins can pass the
```
define "external" ?

danwinship replied:
"not kubenet". external to kubernetes/kubernetes.

danwinship commented:
OK, updated:
- the basic tests will initially run in their own separate periodic job that enables the feature gate
- the connectivity tests will be `[Disruptive]`
- split out the "verify that iptables rules are correct" parts into separately-Skippable tests
- added a few more test cases
